man/tilux.1
.Dd April 9, 2024
.Dt Tilux 1
.Os
.Sh NAME
.Nm tilux
.Nd Command-line interface tool for Linux
.Sh SYNOPSIS
.Bl -tag -width Ds
.It Xo
.Nm
.Op Fl h | \-\-help
.Xc
.El
.Sh DESCRIPTION
.Nm
is a command-line interface (CLI) tool that provides a collection of useful and concise scripts categorized into four folders:
.br
Cryptography, Monitoring, Networking, and System.

.Pp
The available folders and their contents are as follows:

.Bl -bullet
.It
Cryptography: ccrypt, openssl, pyca, and zip with encryption and
.br
decryption options.
.It
Monitoring: CPU processes, memory consumption, and website
.br
availability.
.It
Networking: IP and server pinging.
.It
System: Various system-related scripts including configurations and system information.
.El

.Pp
Note: The options correspond to the folders as follows: 1 (Cryptography), 2 (Monitoring), 3 (Networking), and 4 (System).

.Sh OPTIONS
.Bl -tag -width "-c | -mc"
.It Fl h | \-\-help
Display help information.

.It 1 \-cc \-e
ccrypt encryption.

.It 1 \-cc \-d
ccrypt decryption.

.It 1 \-op \-e
Openssl encryption.

.It 1 \-op \-d
Openssl decryption.

.It 1 \-pc \-e
Pyca encryption.

.It 1 \-pc \-d
Pyca decryption.

.It 1 \-zp \-e
Zip encryption.

.It 1 \-zp \-d
Zip decryption.

.It 2 \-c
Monitoring CPU processes.

.It 2 \-m
Monitoring memory consumption.

.It 2 \-w
Monitoring website (if it's up and if not it sends an email).

.It 3 \-i
IP pinging.

.It 3 \-s
Server pinging.

.It 4 \-c \-gc
Configure git.

.It 4 \-c \-gtd
Gnome terminal profile dump.

.It 4 \-c \-gtl
Gnome terminal profile load.

.It 4 \-f \-bf
Backup folder.

.It 4 \-f \-c
Clean system of logs and trash.

.It 4 \-f \-char
Amount of chars in a file.

.It 4 \-f \-d
Check for duplicate files.

.It 4 \-f \-del
Delete file or directory.

.It 4 \-f \-dir
Directory size calculator.

.It 4 \-f \-e
Exists (check if file or dir exists).

.It 4 \-f \-ext
Check file extensions inside a folder.

.It 4 \-f \-f
Check file permissions.

.It 4 \-f \-fd
FD (number of files and folders).

.It 4 \-f \-fl
File updated.

.It 4 \-f \-fs
File size.

.It 4 \-f \-k
Search files by keyword.

.It 4 \-f \-l
Last updated (file).

.It 4 \-img \-ci
Convert image.

.It 4 \-img \-i
Image info.

.It 4 \-img \-ri
Resize image.

.It 4 \-img \-ro
Rotate image.

.It 4 \-i \-bit
Computer bit system.

.It 4 \-i \-ip
IP address.

.It 4 \-i \-k
Kernel and distro version.

.It 4 \-i \-os
Check OS.

.It 4 \-i \-prv
Python and Ruby version.

.It 4 \-i \-r
Root (check if you are root or not).

.It 4 \-i \-s
Display computer space, uptime and number of processes running.

.It 4 \-o \-b
Print text in ASCII art (submodule).

.It 4 \-o \-cdp
Convert doc to pdf.

.It 4 \-o \-def
Default web browser.

.It 4 \-o \-p
Parse date.

.It 4 \-o \-rl
Remove line.

.It 4 \-o \-rt
Resize terminal.

.It 4 \-o \-s
Search specific command.

.It 4 \-p \-cpu
CPU name.

.It 4 \-p \-mc
Multi-core process time.

.It 4 \-p \-oc
Single-core process time.

.It 4 \-p \-ps
List active processes.

.It 4 \-s \-fp
Force poweroff.

.It 4 \-s \-fr
Force reboot.
.El

.Sh AUTHOR
.Nm
was written by Endormi.


RELEASES.md
# Releases

## 0.1.1
- Add help, server pinging and exists
- Add misc.
- Add GitHub templates

## 0.1.0
- Add hostname and IP address
- Add the option to use args

## 0.0.1
- Initial release
- Added CPU and memory monitoring


tilux.rb
#!/usr/bin/ruby

require_relative "tools/col"
require_relative "tools/def"
require_relative "tools/catch_exception"


version="v.0.2.5"
puts "
ooooooooooooo ooooo ooooo        ooooo     ooo ooooooo  ooooo
8    888    8  888   888          888       8    8888    d8
     888       888   888          888       8     Y888..8P
     888       888   888          888       8       8888
     888       888   888          888       8     .8PY888.
     888       888   888       o   88.    .8     d8    888b
    o888o     o888o o888ooooood8     YbodP     o888o  o88888o
".green
print "version #{version}".light_yellow
print " by Endormi ".white
link="\e]8;;https://github.com/endormi/tilux\a(github.com/endormi/tilux)\e]8;;\a"
puts link.white

sleep 1

prompt="tilux~# ".green


def sys(file)
  system("clear")
  system(file)
end

def func_to_run(func)
  system("clear")
  system(func)
end


if (ARGV.empty?)
  puts "\nWhat script do you want to run?\n\n".white
  system("bash tools/funcs info")
  system("bash tools/funcs cryptography")
  system("bash tools/funcs monitoring")
  system("bash tools/funcs networking")
  system("bash tools/funcs sys")
  #system("bash tools/funcs misc")
  system("bash tools/funcs help2")

  list_of_commands="Info, Cryptography, Monitoring, Networking, Sys && --help"

  puts "Choices: #{list_of_commands}"
  print "#{prompt} "
  c = gets.chomp.to_s.downcase

  case c
    when "info"
      sys("./info.rb")

    when "cryptography", "crypto"
      func_to_run("bash tools/funcs cryptography")
      print "#{prompt} "
      input = gets.chomp.to_s.downcase

      case input
        when "1", "-c", "--c", "-C", "--C"
          # TODO
          print "#{prompt} "
          i = gets.chomp.to_s.downcase

          ccrypt_options(i)

        when "2", "-g", "--g", "-G", "--G"
          # TODO
          print "#{prompt} "
          i = gets.chomp.to_s.downcase

          gpg_options(i)

        when "3", "-o", "--o", "-O", "--O"
          # TODO
          print "#{prompt} "
          i = gets.chomp.to_s.downcase

          openssl_options(i)

        when "4", "-p", "--p", "-P", "--P"
          # TODO
          print "#{prompt} "
          i = gets.chomp.to_s.downcase

          pyca_options(i)

        when "5", "-z", "--z", "-Z", "--Z"
          # TODO
          print "#{prompt} "
          i = gets.chomp.to_s.downcase

          zip_options(i)

        else
          # TODO
          puts "Invalid choice"
        end

    when "monitoring"
      func_to_run("bash tools/funcs monitoring")
      print "#{prompt} "
      input = gets.chomp.to_s.downcase

      monitoring_options(input)

    when "networking", "net"
      func_to_run("bash tools/funcs networking")
      print "#{prompt} "
      input = gets.chomp.to_s.downcase

      networking_options(input)

    when "sys", "system"
      func_to_run("bash tools/funcs sys")
      print "#{prompt} "
      input = gets.chomp.to_s.downcase

      sys_options(input)

    #when "misc"
    #  puts "misc"

    when "help", "--help", "--h"
      system("bash tools/funcs help")
    else
      puts("\nNot a valid choice!")
      puts "Use #{list_of_commands}"
  end

else
  case ARGV[0]
    when "info"
        sys("./info.rb")

    when "cryptography", "crypto"
      case ARGV[1]
        when "1", "-c", "--c", "-C", "--C"
          ccrypt_options(ARGV[2])

        when "2", "-g", "--g", "-G", "--G"
          gpg_options(ARGV[2])

        when "3", "-o", "--o", "-O", "--O"
          openssl_options(ARGV[2])

        when "4", "-p", "--p", "-P", "--P"
          pyca_options(ARGV[2])

        when "5", "-z", "--z", "-Z", "--Z"
          zip_options(ARGV[2])
        else
          # TODO
          puts "Invalid choice"
      end

    when "monitoring"
      monitoring_options(ARGV[1])

    when "networking", "net"
      networking_options(ARGV[1])

    when "sys", "system"
      sys_options(ARGV[1])

    #when "misc"
    #  puts("misc")

    when "help", "--help", "--h"
      system("bash tools/funcs help")
    else
      puts("Not a valid choice!")
  end
end
